substrate/frame/staking-async/src/session_rotation.rs

```diff
@@ -582,8 +567,11 @@ impl<T: Config> Rotator<T> {
 			Some((time, id)) if Some(id) == current_planned_era => {
 				// We rotate the era if we have the activation timestamp.
 				Self::start_era(active_era, starting, time);
-				// accumulate pruning weight.
-				weight.saturating_accrue(T::WeightInfo::prune_era(ValidatorCount::<T>::get()));
+				// accumulate pruning weight. Let's go for the most pessimistic case (claimed
+				// rewards according to benchmarking)
+				weight.saturating_accrue(T::WeightInfo::prune_era_claimed_rewards(
```

Comment:
nope, we should not have the pruning weight here anymore, right? nothing happens in `end_session`